indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -702,12 +711,12 @@ private TaskLockPosse createNewTaskLockPosse(LockRequest request)
    * for the given requests. Updates the holder with the allocated segment if
    * the allocation succeeds, otherwise marks it as failed.
    */
-  @VisibleForTesting
-  void allocateSegmentIds(
+  private void allocateSegmentIds(
       String dataSource,
       Interval interval,
       boolean skipSegmentLineageCheck,
-      Collection<SegmentAllocationHolder> holders
+      Collection<SegmentAllocationHolder> holders,
+      boolean skipSegmentPayloadFetchForAllocation
```

Comment:
Please rename this argument everywhere.

You can either call it `reduceMetadataIO` same as the config or `fetchRequiredSegmentsOnly` or something.

```suggestion
      boolean reduceMetadataIO
```